dfttopif/parsers/pwscf.py

```diff
@@ -51,39 +68,27 @@ def test_if_from(self, directory):
 
     def get_version_number(self):
         '''Determine the version number from the output'''
-        line = self._get_line('Program PWSCF', self.outputf)
-        version = " ".join(line.split('start')[0].split()[2:]).lstrip('v.')
-        # return Value(scalars=version)
-        return version
+        return self.settings["version"]
 
     def get_xc_functional(self):
         '''Determine the xc functional from the output'''
-        # the xc functional is described by 1 or 4 strings
-        # SLA  PZ   NOGX NOGC ( 1 1 0 0 0)
-        # SLA  PW   PBX  PBC (1434)
-        # PBE ( 1  4  3  4 0 0)
-        # PBE0 (6484)
-        # HSE (14*4)
-        xcstring = self._get_line('Exchange-correlation', self.outputf).split()[2:6]
-        for word in range(4):
-            if xcstring[word][0] == '(':
-                xcstring = xcstring[:word]
-                break
-        return Value(scalars=" ".join(xcstring))
+        return Value(scalars=" ".join(self.settings["exchange-correlation"]))
 
     def get_cutoff_energy(self):
         '''Determine the cutoff energy from the output'''
-        cutoff = self._get_line('kinetic-energy cutoff', self.outputf).split()[3:]
-        return Value(scalars=float(cutoff[0]), units=cutoff[1])
+        return Value(
+            scalars=self.settings["kinetic-energy cutoff"],
+            units=self.settings['kinetic-energy cutoff units']
+        )
 
     def get_total_energy(self):
         '''Determine the total energy from the output'''
-        with open(os.path.join(self._directory, self.outputf)) as fp:
-            # reading file backwards in case relaxation run
-            for line in reversed(fp.readlines()):
-                if "!" in line and "total energy" in line:
-                    energy = line.split()[4:]
-                    return Property(scalars=float(energy[0]), units=energy[1])
+        if 'total energy' in self.settings:
+            return Property(
+                scalars=self.settings['total energy'],
+                units=self.settings['total energy units']
```

Comment:
Check for total energy units being present? Or will it be true that it is always present if total energy is?

Reply:
`dftparse` always pulls those out together.  They come from a line that looks like:
```
     total energy              =    -723.28424107 Ry
```

Reply:
It looks like that pattern is common for all of the properties that are created. It's true that all properties will be coupled with a set of units in dftparse?